sktime/param_est/compose.py

```diff
@@ -301,3 +302,120 @@ def get_test_params(cls, parameter_set="default"):
             params = params + [{"transformers": [t1, t2], "param_est": p}]
 
         return params
+
+
+class FunctionParamFitter(BaseParamFitter):
+    r"""Constructs a parameter fitter from an arbitrary callable.
+
+    A FunctionParamFitter forwards its X argument to a user-defined
+    function (or callable object) and sets the result of this function
+    to the ``param`` attribute. This can be useful for stateless
+    estimators such as simple conditional parameter selectors.
+
+    Note: If a lambda function is used as the ``func``, then the
+    resulting estimator will not be pickleable.
+
+    Parameters
+    ----------
+    param : str
+        The name of the parameter to set.
+    func : callable (X: X_type, **kwargs) -> Any
+        The callable to use for the parameter estimation. This will be
+        passed the same arguments as estimator, with args and kwargs
+        forwarded.
+
+    See Also
+    --------
+    sktime.param_est.plugin.PluginParamsForecaster :
+        Plugs parameters from a parameter estimator into a forecaster.
+    sktime.forecasting.compose.MultiplexForecaster :
+        MultiplexForecaster for selecting among different models.
+
+    Examples
+    --------
+    This class could be used to contruct a parameter estimator that
+    selects a forecaster based on the input data's length. The
+    selected forecaster can be stored in the `selected_forecaster_`
+    attribute, which can be then passed down to a
+    ``MultiplexForecaster`` via a ``PluginParamsForecaster``.
+
+    >>> import numpy as np
+    >>> param_est = FunctionParamFitter(
+    ...     param="selected_forecaster",
+    ...     func=(
+    ...         lambda X, threshold: "naive-seasonal"
+    ...         if len(X) >= threshold
+    ...         else "naive-last"
+    ...     ),
+    ...     kw_args={"threshold": 7},
+    ... )
+    >>> param_est.fit(np.asarray([1, 2, 3, 4]))
```

Comment:
failure is due to how doctest work, it compares the printouts against what is below the line. Here, it expects `FucntionParamFitter(...)`, as the return of `fit` is self. You can use the ellipsis arg `...` to catch the param printout which may format differently by operating system.

Reply:
Good catch! Fixed ✅